velox/functions/lib/window/NthValueBase.h

```diff
@@ -96,6 +64,23 @@ class NthValueFunction : public exec::WindowFunction {
     partitionOffset_ += numRows;
   }
 
+ protected:
+  void initializeConstantOffset(
+      const std::optional<vector_size_t>& constantOffset) {
+    if (constantOffset.has_value()) {
+      VELOX_USER_CHECK_GE(
+          constantOffset.value(), 1, "Offset must be at least 1");
+      constantOffset_ = constantOffset;
+    } else {
+      isConstantOffsetNull_ = true;
+    }
+  }
+
+  void initializeOffsetVector(const exec::WindowFunctionArg& arg) {
+    offsetIndex_ = arg.index.value();
+    offsets_ = BaseVector::create<FlatVector<int64_t>>(BIGINT(), 0, pool());
```

Comment:
It looks like there is an implicit assumption that offset type is BIGINT unless it is constant. Ideally, we would remove this assumption, but if we can't, let's make it explicit by adding a VELOX_CHECK for arg.type() == BIGINT() and providing class-level comments which explain this limitation.

Reply:
I moved the Bigint related initialization logic to the Presto NthValue class, and added the type check there also, Thanks.